CMakeLists.txt

```diff
@@ -0,0 +1,225 @@
+ADD_CUSTOM_COMMAND(
+    OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/build_meta_data.c
+            ${CMAKE_CURRENT_BINARY_DIR}/_build_meta_data.c
+    COMMAND ${CMAKE_COMMAND} -P
+            ${CMAKE_CURRENT_SOURCE_DIR}/cmake/BuildMetaData.cmake
+    COMMENT "Generating build meta data.."
+)
+
+##
+## build libtdnfclient.so
+##
+
+### External dependency: rpm-devel
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(RPM REQUIRED rpm)
+include_directories(${RPM_INCLUDE_DIRS})
+
+### External dependency: libsolv
+find_package(LibSolv REQUIRED ext)
+
+### External dependency: libcurl
+find_package(CURL REQUIRED)
+
+set(LIB_TDNF_CLIENT tdnfclient)
+
+list(APPEND LIB_TDNF_COMMMON_SRC
+    "${CMAKE_BINARY_DIR}/build_meta_data.c"
+    "common/memory.c"
+    "common/strings.c"
+    "common/configreader.c"
+    "common/setopt.c"
+    "common/utils.c"
+)
+
+list(APPEND LIB_TDNF_SOLV_SRC
+    "solv/tdnfpool.c"
+    "solv/tdnfrepo.c"
+    "solv/tdnfquery.c"
+    "solv/tdnfpackage.c"
+)
+
+list(APPEND LIB_TDNF_CLIENT_SRC
+    "client/api.c"
+    "client/clean.c"
+    "client/client.c"
+    "client/config.c"
+    "client/goal.c"
+    "client/gpgcheck.c"
+    "client/init.c"
+    "client/packageutils.c"
+    "client/repo.c"
+    "client/remoterepo.c"
+    "client/repolist.c"
+    "client/repoutils.c"
+    "client/resolve.c"
+    "client/rpmtrans.c"
+    "client/updateinfo.c"
+    "client/utils.c"
+    "client/validate.c"
+)
+
+add_library(${LIB_TDNF_CLIENT} SHARED
+    ${LIB_TDNF_CLIENT_SRC}
+    ${LIB_TDNF_SOLV_SRC}
+    ${LIB_TDNF_COMMMON_SRC}
+)
+
+set_target_properties(${LIB_TDNF_CLIENT} PROPERTIES
+    OUTPUT_NAME "tdnf"
+    VERSION "2.0.0"
+    SOVERSION "2"
+)
+
+install(TARGETS ${LIB_TDNF_CLIENT} LIBRARY DESTINATION lib COMPONENT library)
+
+##
+## build libtdnfcli.so
+##
+
+set(LIB_TDNF_CLI tdnfcli)
+
+list(APPEND LIB_TDNF_CLI_SRC
+    "tools/cli/lib/api.c"
+    "tools/cli/lib/help.c"
+    "tools/cli/lib/installcmd.c"
+    "tools/cli/lib/options.c"
+    "tools/cli/lib/output.c"
+    "tools/cli/lib/parseargs.c"
+    "tools/cli/lib/parsecleanargs.c"
+    "tools/cli/lib/parselistargs.c"
+    "tools/cli/lib/parserepolistargs.c"
+    "tools/cli/lib/parseupdateinfo.c"
+    "tools/cli/lib/updateinfocmd.c"
+)
+
+add_library(${LIB_TDNF_CLI} SHARED ${LIB_TDNF_CLI_SRC})
+
+set_target_properties(${LIB_TDNF_CLI} PROPERTIES
+    VERSION "2.0.0"
+    SOVERSION "2"
+)
+
+install(TARGETS ${LIB_TDNF_CLI} LIBRARY DESTINATION lib COMPONENT library)
+
+##
+## tdnf binary
+##
+
+set(TDNF_BIN tdnf)
+
+list(APPEND TDNF_BIN_SRC
+    "tools/cli/main.c"
+)
+
+add_executable(${TDNF_BIN} ${TDNF_BIN_SRC})
+
+target_link_libraries(${TDNF_BIN} PUBLIC
+    ${LIB_TDNF_CLIENT}
+    ${LIB_TDNF_CLI}
+    ${RPM_LIBRARIES}
+    ${LibSolv_LIBRARIES}
+    ${CURL_LIBRARIES}
+)
+
+install(TARGETS ${TDNF_BIN} RUNTIME DESTINATION bin COMPONENT binary)
+
+##
+## Other static resources
+##
+
+install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/resources/tdnf.conf"
+    DESTINATION "/ect/tdnf"
```

Comment:
Typo. /etc/

Reply:
Which one? `tdnf.conf` is installed inside `/etc/tdnf/`.

Reply:
DESTINATION "/ect/tdnf" 
Instead of /etc/ its mentioned as /ect/

Reply:
Ahh... Sorry. Got it now.